### ISSUE TITLE
plugin not visible in revit 2024

### ISSUE THREAD

**Issue by RichardPinka** [jeremytammik/RevitLookup#281]
### RevitLookup version

2025.0.8

### Steps to reproduce

actually, i am not sure, wether only the newest version of plugin shall be installed for all versions of revit i have installed. 

### ✔️ Expected Behavior

plugin will run and i will see its buttons in revit
but in revit 2024 it is not working. in 2022 works. 

### ❌ Actual Behavior

actually, there is some error message 
![image](https://github.com/user-attachments/assets/61f227d9-3c8d-4388-8554-809a44eb5534)

**Comment by Nice3point** [comment]
Please run the [DependenciesReport](https://github.com/user-attachments/files/16818406/DependenciesReport.zip) tool and send a report https://github.com/jeremytammik/RevitLookup/issues/269#issuecomment-2323309590

**Comment by Nice3point** [comment]
and why do you specify version 2025.0.8 but describe 2022, 2024 ?

**Comment by RichardPinka** [comment]
just a kind update that  You speak with a person witout specialization in the IT and programming. that´s why maybe  confused answers : 
dependencies report => i double clicked on the file by mouse , but nothing happened. i think it should do something but nothing happens.  or is there particular way how to use it ?

and why do you specify version 2025.0.8 but describe 2022, 2024 ?  => well , i am get used to install one plugin for all versions of revits. Is this working also the same with Revit lookup plugin ? or  for Revit 2023 i have to have installed particular msi installer  and on Revit 2024 another one ?

**Comment by Nice3point** [comment]
Correct, you need to install a separate RevitLookup version for each version separately. This is specified in each release and Jeremy writes it in every blog post. 
https://github.com/jeremytammik/RevitLookup/wiki/Versions
![изображение](https://github.com/user-attachments/assets/d5944dd2-f71d-4021-90d6-9c16c5c1519d)

> i double clicked on the file by mouse , but nothing happened
You don't get the console showing up like the screenshot to the post? https://github.com/jeremytammik/RevitLookup/issues/269#issuecomment-2323309590

**Comment by RichardPinka** [comment]
Thank You for fast response. installation of revit version : well, from the link of Versions you sent me, 
is actually Revit 2024 usage connected with version 2025.0.8     

on the link is this : 
![image](https://github.com/user-attachments/assets/98dd492d-ed55-44e6-a089-caf486982bd2)
and below , row for Revit 2024 
you are directed onto this page : 
https://github.com/jeremytammik/RevitLookup/releases/tag/2025.0.8

so I guess i have to install version 2025.0.8   if i want to use it in Revit 2024. 
but if it is like this, it is in contradiction with what you just wrote (because i would interpret your message like i shall have installed 
version 2024.1.8 of RevitLookup for Revit 2024   and e.g -2023.2.7 Revit lookup for Revit 2023.

but when i install 2025.0.8 version i  have that error message shown before. huh... 

If its not like that, i am sorry but i am confused and in link you showed me, are not more information. 

to the other part of queston : 
You don't get the console showing up like the screenshot to the post?   

uhh. no, no.. no window shows up within my windows. I havent seen anything like this on my computer (by double clicking onto the application DependenciesReport) . 
Do you want me to sent you at least the speccy output of the parameters of my windows 10 with all its settings ?

**Comment by Nice3point** [comment]
> so I guess i have to install version 2025.0.8 if i want to use it in Revit 2024. but if it is like this, it is in contradiction with what you just wrote (because i would interpret your message like i shall have installed version 2024.1.8 of RevitLookup for Revit 2024 and e.g -2023.2.7 Revit lookup for Revit 2023.

That's right, you're navigating to the latest Release 2025.0.8 as listed on the Versions page. And download version 2024.1.8 for the 2024 version of Revit:
![изображение](https://github.com/user-attachments/assets/519a1a02-8205-4843-bd55-33dc42c36342)

 The current releases contain installers for Revit versions 2021 to 2025, each one needs to be installed separately.

You might be confused by the release name 2025.0.8 and think that this release is only for Revit version 25, but it is not. Perhaps we should change the release names, but what are the options? 🤔

> uhh. no, no.. no window shows up within my windows. I havent seen anything like this on my computer

Strange. Please try the next steps:
1. Open the explorer and navigate to the folder with your file. Type `cmd` in the address line end press `Enter`
![изображение](https://github.com/user-attachments/assets/61bb61d5-0bc4-4e63-a163-225389d86f07)
2. Type the file name  and press enter
![изображение](https://github.com/user-attachments/assets/9957e6d9-0e82-421c-9097-dd14c5d06a5e)

**Comment by RichardPinka** [comment]
Hello, thanx for navigation. 
what are to possible options ?   to be more pragmatical, why not to name particular installation files like 
Revit lookup for Revit 2023_v3.8
Revit lookup for Revit 2025_v0.8  
etc... 

after your suggestion, there was message : 
![image](https://github.com/user-attachments/assets/9a7c5c33-006a-4c7a-8889-a5ea445aa701)

after installing that, the next error window
![image](https://github.com/user-attachments/assets/129ae6a1-16af-4b20-bbc2-9ca164d94e71)

it seems the Diroots One plugin  has some message... 

but as it is one of the best plugins on this planet, solving many usecases , that revit developers have not been willing to develop, i barely tend to uninstall it.

**Comment by Nice3point** [comment]
> it seems the Diroots One plugin has some message...

Try this file with a workaround to the DiRoots issue
[DependenciesReport.zip](https://github.com/user-attachments/files/16845557/DependenciesReport.zip)

**Comment by RichardPinka** [comment]
Hello dear Roman, thank You for support. Here is dependenciees report file.. it also asks me to "try to fix dependencies and upgrade to the latest version, what shall i do ? 

[DependenciesReport-2024-09-04.txt](https://github.com/user-attachments/files/16868914/DependenciesReport-2024-09-04.txt)

**Comment by Nice3point** [comment]
@RichardPinka 

The script found only one conflict with RevitLookup, I hope you don't have any plugins that are installed in non-default directories. 

Can you try disabling RevitPythonShell temporarily and see if Lookup starts?

If that doesn't help, it would be great if you could take a look at the table, and for version 2024 let me know if all installed plugins are in the table, or if you have more?

![изображение](https://github.com/user-attachments/assets/9d7ecb30-87fd-4f96-bbc5-e74190a34c96)

**Comment by RichardPinka** [comment]
Hello, @Nice3point Well i usually dont care where the plugin will be placed.. usually I am not changing default destinations.  
Problem persists after uninstalling the revit python shell. 
also, when i turned-off all plugins but just revit lookup was turned on , the problem has not been sorted out and the same error window appeared after starting revit. 
list of actually used plugins : 
![image](https://github.com/user-attachments/assets/eb11793a-59c6-4d1e-b2a3-dc2a75c7c98a)
2nd page
![image](https://github.com/user-attachments/assets/7f8c2bcc-4cb9-4db1-ad65-78f723013ec8)
3rd page
![image](https://github.com/user-attachments/assets/6da6e693-b499-47ec-beaf-6b97978637f5)

**Comment by Nice3point** [comment]
Thanks for the additional information, I will investigate and improve the script, hopefully we will get to the truth and which plugin is causing RevitLookup to fail

**Comment by Nice3point** [comment]
@RichardPinka could you run a new script version? send me a txt report
[DependenciesReport.zip](https://github.com/user-attachments/files/16901170/DependenciesReport.zip)

**Comment by RichardPinka** [comment]
@Nice3point  yes, here it is
[DependenciesReport-2024-09-07.txt](https://github.com/user-attachments/files/16917289/DependenciesReport-2024-09-07.txt)

**Comment by Nice3point** [comment]
@RichardPinka In this variant I have added detailed information about dependencies. Send me a report to see what versions you have installed in general:
[DependenciesReport.zip](https://github.com/user-attachments/files/16922414/DependenciesReport.zip)

To be honest, I have no idea why it might not work for you. You still don't have the 2024 version working, right? If you really claim to have disabled all other plugins, this case is difficult to solve as RevitLookup works out of the box

**Comment by RichardPinka** [comment]
[DependenciesReport-2024-09-09.txt](https://github.com/user-attachments/files/16933016/DependenciesReport-2024-09-09.txt)
dependencies report uploaded.

**Comment by jeremytammik** [comment]
> some tool to help a human read Revit journal file?

check out https://thebuildingcoder.typepad.com/blog/2022/05/analysis-of-macros-journals-and-add-in-manager.html#4

**Comment by RichardPinka** [comment]
Thank You @jeremytammik , i will check them. anyway it seems my trouble with lagging of the Revit (after i click back into it from other software, e.g. excel, or chrome) has been sorted out by increase of virtual memory of my C drive. 
@Nice3point  i run dependencies report and this was the answer : 
![image](https://github.com/user-attachments/assets/27d0f152-564c-4f88-a27a-275eb2210a93)

i installed the revit Lookup single user 2024.1.8  local built, but without any change.

**Comment by Nice3point** [comment]
@RichardPinka interesting story, you have installed Dotnet but you don't see the system dependency. Never mind, in that case I have built a Native build especially for you, it doesn't need Dotnet installed and doesn't need any dependencies
[DependenciesReport.zip](https://github.com/user-attachments/files/16961962/DependenciesReport.zip)

**Comment by RichardPinka** [comment]
@Nice3point  thank You for your effort. 
[DependenciesReport-2024-09-15.txt](https://github.com/user-attachments/files/17004909/DependenciesReport-2024-09-15.txt)

regarding the update, You were right, usually I'm counting on automatic update notifications and there was slightly newer update on cloud, which update notifier did not show for download+install. 

installation version before was : 
<img width="666" alt="20240915_BEFORE" src="https://github.com/user-attachments/assets/89921e17-7cf6-4ef8-a13c-3f2dd659a0b0">

installation version after udpate  : 
<img width="624" alt="20240915_AFTER" src="https://github.com/user-attachments/assets/4a9d237c-2369-4cd1-b128-1a3bf0080316">

**Comment by Nice3point** [comment]
Great, did I understand correctly that the Revit update fixed the issue with RevitLookup launching?

**Comment by RichardPinka** [comment]
... to finish my answer (after installing update and rebooting computer) i reinstalled also the RevitLookup plugin, by the file you sent me in this conversation. 
error persisted :
![image](https://github.com/user-attachments/assets/92b2f332-5726-4085-990e-a5a8b067f5ba)
here is the dependencies report used second time
[DependenciesReport-2024-09-15_2.txt](https://github.com/user-attachments/files/17005262/DependenciesReport-2024-09-15_2.txt)

... after accepting "try to fix dependencies and upgrade to the latest version? => Y 
here is report 
![image](https://github.com/user-attachments/assets/57bfda6c-956a-4659-8c66-532a792104b7)
(after turned off the opened revit software ... finishing it by running script for second time )
![image](https://github.com/user-attachments/assets/eef25bae-0421-4de2-93ba-1a4d515d45f9)

after this updates (without rebooting) the revit agan showed error. but maybe not the same ? 
![image](https://github.com/user-attachments/assets/ee9aab34-1e8d-4583-8dc3-765e11282809)

**Comment by Nice3point** [comment]
@RichardPinka no way, let's try scanning the dependencies loaded in Revit
[AssembliesReport-1.0.0-SingleUser.zip](https://github.com/user-attachments/files/17006045/AssembliesReport-1.0.0-SingleUser.zip)

- Install AssembliesReport addin
- Open Revit where Lookup is not started.
- Run AssembliesReport.
- A file will be created, send it here. This will be an easier way to trace dependencies loading

![изображение](https://github.com/user-attachments/assets/85982c9b-6080-4a12-a2aa-18193d0f29ee)

**Comment by RichardPinka** [comment]
hello, i think after installation of the plugin new error window after starting revit occured (thus 2 together after starting) 
![image](https://github.com/user-attachments/assets/8cada511-2fb0-4679-9333-aade374eb4dc)
second one
![image](https://github.com/user-attachments/assets/f7772220-43f6-4dda-9668-10939898b804)

after running the assemblies report
![image](https://github.com/user-attachments/assets/2af8670b-03b7-4b85-a971-51fcae5e88d1)


and i rather send you also journal files from the session i test this ... maybe it brings you new info

[journal.0227.txt](https://github.com/user-attachments/files/17013200/journal.0227.txt)
[journal.0227.worker1.log](https://github.com/user-attachments/files/17013203/journal.0227.worker1.log)

**Comment by Nice3point** [comment]
Illegal form🤔 Try this build, we can skip invalid assemblies
[AssembliesReport-1.0.1-SingleUser.zip](https://github.com/user-attachments/files/17013859/AssembliesReport-1.0.1-SingleUser.zip)

**Comment by RichardPinka** [comment]
ok, now it exported a file :D 
[AssembliesReport-2024-09-17.txt](https://github.com/user-attachments/files/17024602/AssembliesReport-2024-09-17.txt)

**Comment by Nice3point** [comment]
@RichardPinka finally 😂

![изображение](https://github.com/user-attachments/assets/045a1ff2-72f7-4300-b9d5-ff1d899ef99e)

I compared your dependencies with mine and found this, Armacell.RVT plugin ships with Serilog version 2.24, Lookup includes the newer 4.0. Can you try disabling Armacell.RVT and see if RevitLookup works ?

**Comment by RichardPinka** [comment]
Ok, i uninstalled the Armacell plugin.  but after running revit, there was not revit lookup visible even it has been installed. 

and it seems RevitLookup is there and working,hurray ! and Thank You for support  !

(by the way, only purpose for this Armacell plugin was, that it was able to put "douple layered insulation" for pipes, which we designed onto pipings in exteriors" . but unfortunatelly another cool function which we would welcome was missing : distinguish, wether the piping is inside internal walls, or in shafts, or above ceiling, which are three different rules where three different logics of insulation applying . unfortunatelly, basic native Revit functions are always forgetting the MEP engineers needs) 


it seemed to me a very difficult to find source of issue which i would expect should be somewere in first error window which opens.. but maybe it is not possible. 

And now I can dig into my main goal which studying material and testing for prompting room/wall geometry for energy computation purposes by starting studying  Jermemys posts related with Building Geometry (in Jeremy´s words: Scott Conover's brilliant class held at Autodesk University 2009 from Jeremys posts...   unfortunatelly, the class material is gone, and also i am trying the Python way 

thank You agan and

**Comment by jeremytammik** [comment]
@RichardPinka , in reply to, unfortunatelly, basic native Revit functions are always forgetting the MEP engineers needs... one option you have is to raise a wishlist item for desired functionality in the [Revit Idea Station](https://forums.autodesk.com/t5/revit-ideas/idb-p/302).

**Comment by Nice3point** [comment]
@RichardPinka 

I've added support for scanning all known paths. Now the script finds conflicts with built-in plugins, and especially importantly with downloaded plugins from Autodesk store.

![изображение](https://github.com/user-attachments/assets/77e9b9e5-55e5-4d7f-ba7c-f9960d327499)

Detailed description: https://github.com/jeremytammik/RevitLookup/issues/269#issuecomment-2323309590

I tried the Armacell download. After fixing the dependencies, RevitLookup and all other plugins continue to work . However, Armacell gives an error. This is because the developers of this plugin use reflection (maybe to load third party dlls?), and the wrong way to load dependencies is their fault, so the only thing to do is to enable this plugin as needed, and wait for Armacell developers to fix it.

Thanks so much again for testing, we have come a long way, hopefully this will help others in the future 🕊️
